homeassistant/components/zwave/__init__.py

```diff
@@ -492,6 +501,31 @@ def set_config_parameter(service):
                      "with selection %s", param, node_id,
                      selection)
 
+    def set_indicator(service):
+        """Set the indicator value of a node."""
+        node_id = service.data.get(const.ATTR_NODE_ID)
+        node = network.nodes[node_id]
+        value = service.data.get(const.ATTR_CONFIG_VALUE)
+        if node.has_command_class(const.COMMAND_CLASS_INDICATOR):
+            for value_id in node.get_values(
+                    class_id=const.COMMAND_CLASS_INDICATOR):
+                node.values[value_id].data = value
+                _LOGGER.info("Node %s indicator set to %d", node_id, value)
+        else:
+            _LOGGER.info("Node %s does not support indicator", node_id)
+
+    def refresh_indicator(service):
```

Comment:
Why would you need this?
Z-wave will automatically fire a veluechanged event and refresh the sensor value immideatly.
If that is not the case there is already the option to set polling, and even if that does not work, we have manual services to refresh values: `refresh_entity` and `refresh_node`

Reply:
Not all devices inform the controller when the indicator value changes. With the Cooper Scene Controller, pressing the buttons will change the led state and the indicator value, but the controller doesn't know to refresh. Also, I tried using the existing `refresh_` services, but because they refresh everything about the node, it seemed to take longer. I didn't do any scientific testing though, so it could have been confirmation bias. In general, I consider it to be best practice to minimize the amount of Z-Wave traffic that is generated to avoid latency.

I didn't even consider polling because I will be triggering automations based on the value.

Reply:
Then I would suggest adopting this:
https://github.com/home-assistant/home-assistant/blob/994b829cb4b706d0103598b194da9a36910cd2f5/homeassistant/components/zwave/__init__.py#L963
Into an available service. This will eneable users to refresh any value on the zwave network.

Reply:
Just to make sure I am understanding correctly, you are saying it should be a generic service that allows you to specify the value you would like to refresh. What would would be passed to specify the value to refresh on a node? `class_id`? `value_id`?

Reply:
Correct. The `value_id` is the identifier to find a certain value i.e. the indicator. 
This can be found either via `unique_id`(value_id is the part after the `node_id`) or directly from `value.value_id`

Reply:
Ok. Makes sense. I will update the PR.